python-avd/pyavd/_anta/input_factories/router_path_selection.py

```diff
@@ -38,26 +38,32 @@ def create(self) -> list[VerifySpecificPath.Input] | None:
             for interface in path_group.local_interfaces:
                 # Get the source IP address for the local interface
                 if interface.name.startswith("Ethernet") and interface.name in self.structured_config.ethernet_interfaces:
-                    ip_address = self.structured_config.ethernet_interfaces[interface.name].ip_address
+                    interface_ip_address = self.structured_config.ethernet_interfaces[interface.name].ip_address
                 elif interface.name.startswith("Port-Channel") and interface.name in self.structured_config.port_channel_interfaces:
-                    ip_address = self.structured_config.port_channel_interfaces[interface.name].ip_address
+                    interface_ip_address = self.structured_config.port_channel_interfaces[interface.name].ip_address
                 else:
-                    ip_address = None
+                    interface_ip_address = None
 
-                if ip_address is None:
+                if interface_ip_address is None:
                     self.logger_adapter.debug(LogMessage.INTERFACE_NO_IP, interface=interface)
                     continue
 
-                if ip_address == "dhcp":
+                if interface_ip_address == "dhcp":
                     self.logger_adapter.debug(LogMessage.INTERFACE_USING_DHCP, interface=interface)
                     continue
 
-                source_address = ip_interface(ip_address).ip
+                source_address = ip_interface(interface_ip_address).ip
                 for static_peer in path_group.static_peers:
-                    for destination_address in static_peer.ipv4_addresses:
-                        dps_path = DpsPath(
-                            peer=static_peer.router_ip, path_group=path_group.name, source_address=source_address, destination_address=destination_address
-                        )
-                        all_dps_paths.append(dps_path)
+                    if isinstance(ip_address(static_peer.router_ip), IPv4Address):
+                        for destination_address in static_peer.ipv4_addresses:
+                            dps_path = DpsPath(
+                                peer=IPv4Address(static_peer.router_ip),
+                                path_group=path_group.name,
+                                source_address=IPv4Address(source_address),
```

Comment:
source_address is already an `IPv4Address` instance no?
```suggestion
                                source_address=source_address,
```

Reply:
no it can be IPv6 as well because it came from `source_address = ip_interface(interface_ip_address).ip`.

Reply:
That means for the type checker we could be doing `IPv4Address(<IPv6Address instance>)` which feels wrong.

Maybe we could simply continue if source_address is an instance of IPv6Address? This would never happen because  it is taken from `ip_address` and not `ipv6_address` so it's fine I think.

Reply:
yeah right it taken from ip_address so it will be ipv4 address only. So just doing continue and not any logger.